src/blockdev.rs

```diff
@@ -717,13 +930,279 @@ mod tests {
+            (
+                vec![label("*i*")],
+                vec![
+                    make_part(1, "boot", 1, 384),
+                    make_part(2, "EFI-SYSTEM", 384, 512),
+                    make_part(4, "root", 1024, 2200),
+                    make_part(5, "five", 4096, 5120),
+                    make_part(8, "eight", 6144, 7168),
+                    make_part(9, "nine", 7168, 8192),
+                ],
+                vec![
+                    make_part(5, "five", 4096, 5120),
+                    make_part(8, "eight", 6144, 7168),
+                    make_part(9, "nine", 7168, 8192),
+                ],
+            ),
+            // Missing label, single partition, irrelevant range
+            (
+                vec![
+                    label("six"),
+                    Index(index(7), index(7)),
+                    Index(index(11), None),
+                ],
+                vec![
+                    make_part(1, "boot", 1, 384),
+                    make_part(2, "EFI-SYSTEM", 384, 512),
+                    make_part(4, "root", 1024, 2200),
+                    make_part(7, "seven", 5120, 6144),
+                ],
+                vec![make_part(7, "seven", 5120, 6144)],
+            ),
+            // Partition renumbering
+            (
+                vec![Index(index(4), None)],
+                vec![
+                    make_part(1, "boot", 1, 384),
+                    make_part(2, "EFI-SYSTEM", 384, 512),
+                    make_part(4, "root", 1024, 2200),
+                    make_part(5, "four", 3072, 4096),
+                    make_part(6, "five", 4096, 5120),
+                    make_part(7, "seven", 5120, 6144),
+                    make_part(8, "eight", 6144, 7168),
+                    make_part(9, "nine", 7168, 8192),
+                ],
+                vec![
+                    make_part(4, "four", 3072, 4096),
+                    make_part(5, "five", 4096, 5120),
+                    make_part(7, "seven", 5120, 6144),
+                    make_part(8, "eight", 6144, 7168),
+                    make_part(9, "nine", 7168, 8192),
+                ],
+            ),
+            // No saved partitions
+            (
+                vec![Index(index(15), None)],
+                vec![
+                    make_part(1, "boot", 1, 384),
+                    make_part(2, "EFI-SYSTEM", 384, 512),
+                    make_part(4, "root", 1024, 2200),
+                ],
+                vec![],
+            ),
+            // No filters
+            (
+                vec![],
+                vec![
+                    make_part(1, "boot", 1, 384),
+                    make_part(2, "EFI-SYSTEM", 384, 512),
+                    make_part(4, "root", 1024, 2200),
+                ],
+                vec![],
+            ),
+        ];
+
+        let base = make_disk(512, &base_parts);
+        for (testnum, (filter, expected_image, expected_blank)) in tests.iter().enumerate() {
```

Comment:
Love these tests!

Minor/optional: wouldn't it make more sense to flip the last two tuple elements, e.g. `(filters, expected_blank, expected_image)`? So it'd be the inputs first, then the expected output.